plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -139,35 +142,48 @@ public DefaultGlueColumnStatisticsProvider(AWSGlueAsync glueClient, String catal
     }
 
     @Override
-    public Map<String, HiveColumnStatistics> getPartitionColumnStatistics(Partition partition)
+    public Map<Partition, Map<String, HiveColumnStatistics>> getPartitionColumnStatistics(Collection<Partition> partitions)
```

Comment:
should this be a `Set` vs `Collection`?

Reply:
Could be, but I'd have to change some call sites. The `HiveMetastore#getPartitionStatistics`  method passes in a `List<Partition>`. Using Collection here let me leave that alone without needing to to a copy to a Set.

Reply:
Partition seems immutable already so should be ok to keep as is. Your choice.